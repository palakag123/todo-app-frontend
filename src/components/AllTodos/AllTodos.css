.allTodos-container{
    width: 65%;
    background-image: linear-gradient(white, rgb(23, 200, 246));
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    overflow-y: hidden;
    overflow-x: hidden;
}
#todos-heading{
    color: rgb(87,204,237);
    font-weight: 500;
    font-size: 45px;
}
.new-todo-button{
    margin-top: 50px;
    margin-bottom: 50px;
    background-color: white;
    border: none;
    outline: none;
    border-radius: 7px;
    padding: 10px;
    padding-right: 20px;
    padding-left: 20px;
    font-size: 25px;
    color: rgb(87,204,237);
    font-weight: 700;
}

@media only screen and (max-width: 770px){
    .allTodos-container{
        width: 100vw;
    }
    #todos-heading{
        font-size: 30px;
    }
    .new-todo-button{
        font-size: 20px;
    }
}