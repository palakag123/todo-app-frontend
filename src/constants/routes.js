export const HOME_ROUTE = "/";
export const CREATE_TODO_ROUTE = "/create-todo";
