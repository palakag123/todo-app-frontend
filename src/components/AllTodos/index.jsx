import React from "react";
import PropTypes from "prop-types";
import { useNavigate } from "react-router-dom";
import AllTodoCards from "../AllTodoCards";
import "./AllTodos.css";
import { CREATE_TODO_ROUTE } from "../../constants/routes";

function AllTodos({ todos }) {
  const navigate = useNavigate();

  const onCreateNewTodo = () => {
    navigate(CREATE_TODO_ROUTE);
  };

  return (
    <div className="allTodos-container">
      <h1 id="todos-heading" data-testid="allTodoHeading">
        ALL TO-DOS
      </h1>
      <AllTodoCards todos={todos} data-testid="allTodoCards" />
      <button
        className="new-todo-button"
        type="button"
        onClick={onCreateNewTodo}
        data-testid="newTodoButton"
      >
        CREATE NEW
      </button>
    </div>
  );
}
AllTodos.propTypes = {
  todos: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      todo: PropTypes.string.isRequired,
      // eslint-disable-next-line comma-dangle
    })
  ).isRequired,
};
export default AllTodos;
