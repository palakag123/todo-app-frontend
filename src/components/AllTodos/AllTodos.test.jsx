import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import AllTodos from "./index";
import { CREATE_TODO_ROUTE } from "../../constants/routes";

const mockNavigate = jest.fn();
jest.mock("react-router-dom", () => ({ useNavigate: () => mockNavigate }));

describe("AllTodos", () => {
  const mockTodos = [{ id: 1, todo: "first todo list item" }];
  beforeEach(() => {
    render(<AllTodos todos={mockTodos} />);
  });
  it("should take a snapshot for AllTodos", () => {
    const { asFragment } = render(<AllTodos todos={mockTodos} />);
    expect(asFragment(<AllTodos todos={mockTodos} />)).toMatchSnapshot();
  });
  it("should have the required elements when AllTodos is rendered", () => {
    expect(screen.getByTestId("allTodoHeading")).toBeTruthy();
    expect(screen.getByText("ALL TO-DOS")).toBeTruthy();
    expect(screen.findAllByText("first todo list item")).toBeTruthy();
    expect(screen.getByTestId("newTodoButton")).toBeTruthy();
    expect(screen.getByText("CREATE NEW")).toBeTruthy();
  });
  it("should call mockNavigate function with the correct route value when the button is clicked", () => {
    fireEvent.click(screen.getByTestId("newTodoButton"));
    expect(mockNavigate).toHaveBeenCalledWith(CREATE_TODO_ROUTE);
  });
});
