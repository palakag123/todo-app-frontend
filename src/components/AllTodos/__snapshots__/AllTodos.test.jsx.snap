// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllTodos should take a snapshot for AllTodos 1`] = `
<DocumentFragment>
  <div
    class="allTodos-container"
  >
    <h1
      data-testid="allTodoHeading"
      id="todos-heading"
    >
      ALL TO-DOS
    </h1>
    <div
      class="all-cards-outer-container"
      data-testid="allCards"
    >
      <div
        class="all-cards-container"
        data-testid="allCardsInner"
      >
        <div
          class="todo-card"
          data-testid="todoCard"
        >
          <p
            id="todo-title"
          >
            first todo list item
          </p>
        </div>
      </div>
    </div>
    <button
      class="new-todo-button"
      data-testid="newTodoButton"
      type="button"
    >
      CREATE NEW
    </button>
  </div>
</DocumentFragment>
`;
